src/commands/test/lib/wizard/uitests.ts

```diff
@@ -0,0 +1,153 @@
+
+interface BuildFolder {
+  name: string;
+  path: string;
+}
+
+export default class RunUitestWizardTestCommand extends AppCommand {
+  private _args: CommandArgs;
+
+  constructor(args: CommandArgs, interactiveArgs: string[]) {
+    super(args);
+    this._args = args;
+    this._args.args.push(...interactiveArgs);
+  }
+
+  public async run(client: AppCenterClient, portalBaseUrl: string): Promise<CommandResult> {
+    const searchFolder: Promise<BuildFolder[]> = this.scanFolder();
+
+    if (this._args.args.indexOf("--async") < 0) {
+      const mergeXml: boolean = await this.promptMergeXml();
+      if (mergeXml) {
+        this._args.args.push("--merge-nunit-xml");
+      }
+    }
+
+    const foundFolders: BuildFolder[] = await searchFolder;
+    const folder: string = await this.promptFolder(foundFolders);
+    this._args.args.push("--build-dir", folder);
+    out.text("\nRunning command: appcenter test run uitest " + this._args.args.join(" ") + "\n");
+    return new RunUITestsCommand(this._args).run(client, portalBaseUrl);
+  }
+
+  private async promptMergeXml(): Promise<boolean> {
+    const questions: Questions = [
+      {
+        type: "list",
+        name: "merge",
+        message: "Should the xml files be merged in to the <output.xml> file?",
+        choices: [{
+          name: "Yes",
+          value: "true"
+        }, {
+          name: "No",
+          value: "false"
+        }]
+      }
+    ];
+    const answers: any = await prompt.question(questions);
+    return answers.merge === "true" ? true : false;
+  }
+
+  private async scanFolder(): Promise<BuildFolder[]> {
+    const foundFolders: BuildFolder[] = [];
+    await this.scanRecurse(process.cwd(), foundFolders);
+    return foundFolders;
+  }
+
+  private async scanRecurse(dirname: string, folders: BuildFolder[]) {
+    const dirContent = fs.readdirSync(dirname);
+    for (const dir of dirContent) {
+      const fullDir = path.join(dirname, dir);
+
+      if (fs.lstatSync(fullDir).isDirectory()) {
+        if (dir !== "node_modules") {
+          const dirContents: string[] = fs.readdirSync(fullDir);
+
+          if (dirContents.length > 0) {
+            const configs = dirContents.filter((file) => {
+              return file === "packages.config";
+            });
+            const projects = dirContents.filter((file) => {
+              const extension: string = path.parse(file).ext;
+              return extension === ".csproj" || extension === ".fsproj";
+            });
+            const containsConfigAndProjectFiles: boolean = configs.length > 0 && projects.length > 0;
+
+            if (containsConfigAndProjectFiles) {
+              let containsTools: boolean = true;
+              try {
+                await UITestPreparer.findXamarinUITestNugetDir(fullDir, fullDir);
+              } catch (e) {
+                containsTools = false;
+              }
+
+              if (containsTools) {
```

Comment:
I don't think this logic is correct - the UITest path for `--build-dir` should be the directory containing binary files for your UITest project (so I'd expect it to be something like `/my-proj/bin/Debug` which contains a number of DLL files including Xamarin.UITest.dll, the NUnit dll and the user's test dll).

Reply:
This is a fair point. The check for config or project files should be incompatible with the average Xamarin project. Will do some more testing to confirm.